packages/migrate/src/commands/DbPull.ts

```diff
@@ -173,30 +174,54 @@ Set composite types introspection depth to 2 levels
             ignoreEnvVarErrors: true,
           })
 
+          const firstDatasource = config.datasources[0] ? config.datasources[0] : undefined
+
           if (input.url) {
-            // TODO: ensure that the `input.url` protocol is compatible with the schema provider
-            // to avoid ambiguities in case of schema parsing errors (error code P1012)
+            const providerFromSchema = firstDatasource?.provider
+            // protocolToConnectorType ensures that the protocol from `input.url` is valid or throws
+            const providerFromUrl = protocolToConnectorType(`${input.url.split(':')[0]}:`)
+            const schema = `${this.urlToDatasource(input.url, providerFromSchema)}\n${removeDatasource(rawSchema)}`
+
+            // If the provider is cockroachdb or postgresql, the are compatible
+            // If not they are not compatible and we throw an error
+            if (
+              providerFromSchema &&
+              providerFromUrl &&
+              Boolean(
+                ['postgresql', 'cockroachdb'].includes(providerFromSchema) &&
+                  ['postgresql', 'cockroachdb'].includes(providerFromUrl),
```

Comment:
Can the provider from the URL ever be `cockroachdb`? Isn't it always `postgresql://`?

Reply:
Fixed